src/core_write.cpp

```diff
@@ -12,17 +12,20 @@
 #include <serialize.h>
 #include <streams.h>
 #include <univalue.h>
-#include <util/system.h>
 #include <util/strencodings.h>
+#include <util/system.h>
 
 UniValue ValueFromAmount(const CAmount& amount)
 {
-    bool sign = amount < 0;
-    int64_t n_abs = (sign ? -amount : amount);
-    int64_t quotient = n_abs / COIN;
-    int64_t remainder = n_abs % COIN;
+    static_assert(COIN > 1);
+    int64_t quotient = amount / COIN;
+    int64_t remainder = amount % COIN;
+    if (amount < 0) {
+        quotient = -quotient;
+        remainder = -remainder;
+    }
     return UniValue(UniValue::VNUM,
-            strprintf("%s%d.%08d", sign ? "-" : "", quotient, remainder));
+            strprintf("%s%d.%08d", amount < 0 ? "-" : "", quotient, remainder));
 }
```

Comment:
```C++
UniValue ValueFromAmount(const CAmount& amount)
{
    int64_t quotient = amount / COIN;
    int64_t remainder = std::abs(amount % COIN);
    static_assert(COIN == 100000000);  // required for format-string below
    return UniValue(UniValue::VNUM, strprintf("%d.%08d", quotient, remainder));
}
```

Reply:
That doesn't work because the sign is lost whenever quotient is `0` (there is no negative zero)

Reply:
Aww, it was *so* close to being a simple/elegant solution, but you're right...